src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -803,6 +812,20 @@ public async Task VersionHandshakeAsync(NetworkPeerRequirement requirements, Can
                 // Ask the just-handshaked peer for the peers they know about to aid in our own peer discovery.
                 await this.SendMessageAsync(new GetAddrPayload(), cancellationToken).ConfigureAwait(false);
             }
+            catch (OperationCanceledException e)
```

Comment:
I see this overload of `VersionHandshakeAsync` is only called by the other overload and the `PeerConnector`  where we already handle all these exceptions?

The other `VersionHandshakeAsync` is called by the discovery loop which just "swallows" the exceptions and `ConnectionManager.ConnectAsync` just handles a the generic `Exception`.

I would perhaps just catch and throw? That way the exception bubbles up and we can still dispose the listener.

Reply:
Changed